### PR TITLE
feat: added Geolocation API to get user's location.

### LINKED ISSUE #5: Using Geolocation API to get the users location
Using the Geolocation API to get the users location as the first weather results displayed, instead of just New York City.

Read more [here](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API).

### PR DESCRIPTION
**Pull request by mukhsadr**
**Commits:**

- [x] Added a useEffect in App.js to do a geolocation call using navigator.geolocation.getCurrentPosition
- [x] Once the user allows the share their location, the altitude and longitude is accessible
- [x] Call a fetch with the coordinates to get all the info of the user's location

Closes #5

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -12,6 +13,9 @@ function App() {
   const [results, setResults] = useState(null);
 
   useEffect(() => {
```
Comment (Keshraf) [inline src/App.js:15]: The functionality you are trying to implement should be done in a different useEffect Hook which only runs when the component is first rendered. Your previous implementation was correct. The implementation that you have right now will lead to infinite re-rendering.

**Review by Keshraf — CHANGES_REQUESTED** [review]

File: yarn.lock
[the remark this answers is not in the document]
Comment (mukhsadr) [reply yarn.lock:1]: In the .gitignore file, I added these lines of code but I can still see there are some conflicts.
yarn.lock
package-lock.json
package.json
.env

File: src/App.js (hunk 1 above)
Comment (mukhsadr) [reply src/App.js]: I created a new useEffect Hook and add the blocks of code we implemented a few days ago.

File: yarn.lock
[the remark this answers is not in the document]
Comment (mukhsadr) [reply yarn.lock:1]: I fixed the conflicts by copying all yarn.lock and package.json and pasting them to the code directly. I felt it is not safe way doing so. I saw 12 conflicts with long lines and got confused. I assumed, I committed my changes before I should have added package.json and yarn.lock. What I did in my local machine was to add them in .gitignore file and went to the conflicts and resolved them by copy and past. Please, write your opinion on that. I felt it is not a safe way doing so. Can you write your suggestions, please.

[the remark this answers is not in the document]
Comment (mukhsadr) [reply yarn.lock:1]: I just removed package.json from .gitignore.

File: .gitignore [hunk 2]
```diff
@@ -21,5 +21,8 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+yarn.lock
+package-lock.json
+package.json
```
[the remark this answers is not in the document]
Comment (mukhsadr) [reply .gitignore]: I just removed package.json from .gitignore.

**Review by Keshraf — APPROVED** [review]
LGTM!

**Review by sumana2001 — COMMENTED** [review]
Screenshot 2023-03-17 at 8 35 16 PM
Why is the yarn.lock file committed?
Also the change in package.json doesn't seem to be required

**Comment by mukhsadr** [comment]
@sumana2001 I added yarn.lock to .gitignore to remove from commit.

**Pull request merged**